packages/yarnpkg-core/sources/httpUtils.ts

```diff
@@ -47,32 +50,50 @@ export function getNetworkSettings(target: string, opts: { configuration: Config
     return keyB.length - keyA.length;
   });
 
-  const mergedNetworkSettings: {
-    enableNetwork?: boolean,
-    caFilePath?: PortablePath | null,
-  } = {};
+  type NetworkSettingsType = MapValueToObjectValue<MapValue<ConfigurationValueMap['networkSettings']>>
+  type UndefinableSettings = { [P in keyof NetworkSettingsType]: NetworkSettingsType[P] | undefined; };
+
+  const mergedNetworkSettings: UndefinableSettings = {
+    enableNetwork: undefined,
+    caFilePath: undefined ,
+    httpProxy: undefined,
+    httpsProxy: undefined,
+  };
+
+  const mergableKeys = Object.keys(mergedNetworkSettings) as Array<keyof NetworkSettingsType>;
 
   const url = new URL(target);
   for (const [glob, config] of networkSettings) {
     if (micromatch.isMatch(url.hostname, glob)) {
-      const enableNetwork = config.get(`enableNetwork`);
-      if (enableNetwork !== null && typeof mergedNetworkSettings.enableNetwork === `undefined`)
-        mergedNetworkSettings.enableNetwork = enableNetwork;
-
-      const caFilePath = config.get(`caFilePath`);
-      if (caFilePath !== null && typeof mergedNetworkSettings.caFilePath === `undefined`) {
-        mergedNetworkSettings.caFilePath = caFilePath;
+      for (const key of mergableKeys) {
+        const setting = config.get(key);
+        if (setting !== null && typeof mergedNetworkSettings[key] === `undefined`) {
+          mergedNetworkSettings[key] = setting as any;
+        }
       }
     }
   }
 
-  if (typeof mergedNetworkSettings.caFilePath === `undefined`)
-    mergedNetworkSettings.caFilePath = opts.configuration.get(`caFilePath`);
+  // Apply defaults
+  for (const key of mergableKeys) {
+    if (typeof mergedNetworkSettings[key] === `undefined`) {
+      mergedNetworkSettings[key] = opts.configuration.get(key) as any;
+    }
+  }
+
+  if (process.env.NO_PROXY) {
+    const skippedHosts = process.env.NO_PROXY.split(`,`);
```

Comment:
`httpUtils` shouldn't rely on `process.env` - `NO_PROXY` should be "transformed" into proper settings within the configuration object.